homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,542 @@
+        conf = config.get(DOMAIN, {})
+
+        # Get the IP address that will be passed to the Echo during discovery
+        self.host_ip_addr = conf.get(CONF_HOST_IP)
+        if self.host_ip_addr is None:
+            self.host_ip_addr = util.get_local_ip()
+            _LOGGER.warning(
+                "Listen IP address not specified, auto-detected address is %s",
+                self.host_ip_addr)
+
+        # Get the port that the Hue bridge will listen on
+        self.listen_port = conf.get(CONF_LISTEN_PORT)
+        if not isinstance(type(self.listen_port), int):
+            self.listen_port = DEFAULT_LISTEN_PORT
+            _LOGGER.warning(
+                "Listen port not specified, defaulting to %s",
+                self.listen_port)
+
+        # Get domains that cause both "on" and "off" commands to map to "on"
+        # This is primarily useful for things like scenes or scripts, which
+        # don't really have a concept of being off
+        self.off_maps_to_on_domains = conf.get(CONF_OFF_MAPS_TO_ON_DOMAINS)
+        if not isinstance(self.off_maps_to_on_domains, list):
+            self.off_maps_to_on_domains = DEFAULT_OFF_MAPS_TO_ON_DOMAINS
+
+        # Get whether or not entities should be exposed by default, or if only
+        # explicitly marked ones will be exposed
+        self.expose_by_default = conf.get(CONF_EXPOSE_BY_DEFAULT)
+        if not isinstance(type(self.expose_by_default), bool):
+            self.expose_by_default = DEFAULT_EXPOSE_BY_DEFAULT
+
+        # Get domains that are exposed by default when expose_by_default is
+        # True
+        self.exposed_domains = conf.get(CONF_EXPOSED_DOMAINS)
+        if not isinstance(self.exposed_domains, list):
+            self.exposed_domains = DEFAULT_EXPOSED_DOMAINS
+
+    def set_up_routes(self, app):
+        """Set up Flask routes."""
+        app.route('/description.xml', methods=['GET'])(
+            self.description_xml)
+
+        app.route('/api/<token>/lights', methods=['GET'])(
+            self.hue_api_lights)
+        app.route('/api/<token>/lights/', methods=['GET'])(
+            self.hue_api_lights)
+
+        app.route('/api/<token>/lights/<entity_id>/state', methods=['PUT'])(
+            self.hue_api_put_light)
+
+        app.route('/api/<token>/lights/<entity_id>', methods=['GET'])(
+            self.hue_api_individual_light)
+
+        app.route('/api', methods=['POST'])(
+            self.hue_api_create_user)
+
+    def description_xml(self):
+        """Handle requests for the bridge's description.xml."""
+        from flask import Response
+
+        xml_template = """<?xml version="1.0" encoding="UTF-8" ?>
+<root xmlns="urn:schemas-upnp-org:device-1-0">
+<specVersion>
+<major>1</major>
+<minor>0</minor>
+</specVersion>
+<URLBase>http://{0}:{1}/</URLBase>
+<device>
+<deviceType>urn:schemas-upnp-org:device:Basic:1</deviceType>
+<friendlyName>HA-Echo ({0})</friendlyName>

```

Comment:
Let's make the `friendlyName` more descriptive, like `Home Assistant Bridge ({0})`